apps/x509.c

```diff
@@ -1154,11 +1154,7 @@ static int callb(int ok, X509_STORE_CTX *ctx)
      * X509_STORE_CTX_get_error(ctx) will still be set to
      * DEPTH_ZERO_SELF_....
      */
-    if (ok) {
-        BIO_printf(bio_err,
-                   "Error with certificate to be certified - should be self-signed\n");
-        return 0;
-    } else {
+    if (!ok) {
```

Comment:
Just removing this is wrong. You should add return 1; to the end of the function.

Also the comment above is now incorrect so please fix it.

Reply:
I have added the missing return statement and removed the obsolete comment.